aten/src/ATen/native/Distributions.cpp

```diff
@@ -184,6 +184,18 @@ Tensor _standard_gamma_grad_cpu(const Tensor& self, const Tensor& output) {
   return ret;
 }
 
+Tensor _dirichlet_grad_cpu(const Tensor& x, const Tensor& alpha, const Tensor& total) {
+  Tensor ret = at::empty(x.sizes(), x.options());
+  AT_DISPATCH_FLOATING_TYPES(x.scalar_type(), "_dirichlet_grad_cpu", [&] {
+    CPU_tensor_apply4<scalar_t, scalar_t, scalar_t, scalar_t>(ret, x, alpha, total,
```

Comment:
Doesn't have to be this PR, but at some point it would be good to port this into using TensorIterator (which I think should be doable here.) CC @VitalyFedyunin

Reply:
There are a lot of dispatches in this file that have to ported into using TensorIterator.